DOC: Add a policy about inactive PRs. (#21537)
numpy/numpy: doc/source/dev/reviewer_guidelines.rst

  explain why. If it's not, then it's a good idea to first explain why you
  think the PR is not suitable for inclusion in NumPy and then let a second
  committer comment or close.
- If the PR submitter doesn't respond to your comments for 6 months, move the PR 
  in question to the inactive category with the “inactive” tag attached. At this point,

>   the PR can be closed by a maintainer. If there is any interest in finalizing the PR under consideration, then this can be indicated at any time (without waiting 6 months) by a comment.   

Author: ```suggestion
  the PR can be closed by a maintainer. If there is any interest in finalizing the PR under consideration,
  this can be indicated at any time, without waiting 6 months, by a comment.   
```

  explain why. If it's not, then it's a good idea to first explain why you
  think the PR is not suitable for inclusion in NumPy and then let a second
  committer comment or close.
- If the PR submitter doesn't respond to your comments for 6 months, move the PR 
  in question to the inactive category with the “inactive” tag attached. At this point,

>   by a quorum of at least 2 maintainers, the PR can be closed or finalized by another 

Reviewer: ```suggestion
  by a quorum of at least 2 maintainers, the PR can be closed. If there is any interest in finalizing said PR, then this can be indicated at any time (without waiting 6 months) by a comment.   
```

There was some discussion on the [25-05-2022](https://hackmd.io/76o-IxCjQX2mOXO_wwkcpg?view) community call. As part of this, the rewording above might be more friendly.

>   contributor or maintainer if the proposed changes are deemed to be valuable.

Reviewer 2: I would suggest dropping or rephrasing the `deemed to be valuable` part. In my experience, the value of a PR is just one factor for finishing it for a contributor (if it's not valuable, then I suspect it was already been suggested to be closed, much sooner than waiting for it to get stale). 
And no matter how valuable a PR is, if it's complicated or is not almost ready, it's likely won't get taken over.

Reviewer 3: Six months for non-response by the submitter seems reasonable, but I suspect non-response by the reviewers is at least as common :) This also puts a burden on reviewers to track the relevant PRs. I like the proposal in the other comments to implement  some sort of automatic tracking and closing. We should pin down how that would operate.

Reviewer 2: Yes, it's a question whether the 6 months become more realistic from the reviewer side of there are only <100 PRs open as opposed to 200+.
How many of those stale PRs are stale because they are lost is the noise as opposed to being complicated, or lack of time from their maintainer etc? 
Also, the number of months can be different, I feel the looking at the devstats could help here. E.g is there a time cutoff where a pr is not likely has conflicts? Is there still a good chance for an old pr to get merged?

Author: > Also, the number of months can be different, I feel the looking at the devstats could help here. E.g is there a time cutoff where a pr is not likely has conflicts? Is there still a good chance for an old pr to get merged?

@bsipocz Those are excellent questions! We'll keep them in mind considering/designing automation solutions.

Reviewer 2: After today's discussion, I have second thoughts about automation. I still think that the status quo and manual closing is not really sustainable, but letting the bot loose on the currently open ones would likely cause the unwanted alienating effects. 

Maybe the automation should first run as a dry-run (the bot can do that), and the output processed into a project board for a heavy triage response?
Maybe the conflicting ones should be picked up on first for a comment (can be semi-automated batch comments on those PRs) asking the contributor to rebase, or if it's clearly close to being ready a maintainer to rebase it and finish the PR straight away?

Either case, I feel it cannot go as fast as the close 70 out of the 1000 issues, or even the close 30 out of the 260 PRs. Those were likely cherry-picked easiest cases, where a clear decision could be made, or the contributor was known, and it was known that alienating wouldn't be an issue.

>   contributor or maintainer if the proposed changes are deemed to be valuable.

Author: ```suggestion

```
